multiversx_sdk/core/transaction_computer.py

```diff
@@ -73,12 +73,31 @@ def apply_guardian(self, transaction: Transaction, guardian: Address) -> None:
         transaction.options = transaction.options | TRANSACTION_OPTIONS_TX_GUARDED
         transaction.guardian = guardian
 
+    def apply_guardina_signature(self, transaction: Transaction, guardian: IAccount):
+        if transaction.guardian != guardian.address:
+            raise Exception("The transaction's guardian does not match the provided guardian account.")
+
+        serialized_tx = self.compute_bytes_for_signing(transaction)
+        transaction.guardian_signature = guardian.sign(serialized_tx)
```

Comment:
Now I see, we didn't have `apply_..._signature` in transaction computer before. That is, in transaction computer, we didn't call `sign` before.

It shouldn't be the computer to do this (to touch the signature field). Sorry for saying this before.

Maybe drop these two functions that touch the signature? Or, to have symmetry, have an additional function that says `apply_signature()` - for the actual sender?

Or in `IAccount`, maybe have `sign_transaction()`, `sign_transaction_as_relayer()`, `sign_transaction_as_guardian()`?

:thought_balloon: :see_no_evil:

Reply:
Will drop the `apply_*_signature` methods and we'll see in the future how we can simplify the signing.